src/cubicbez.rs

```diff
@@ -157,7 +157,15 @@ impl ParamCurveArclen for CubicBez {
                 rec(&c0, accuracy * 0.5, depth + 1) + rec(&c1, accuracy * 0.5, depth + 1)
             }
         }
-        rec(self, accuracy, 0)
+
+        // Check if the bezier curve is degenerate, or almost degenerate
+        // A degenerate curve where all points are identical will cause infinite recursion in the rec function (well, until MAX_DEPTH at least) in all branches.
+        const SMALL : f64 = 0.000000001;
```

Comment:
This seems like a good general approach, but I think it would be better if this were keyed off the accuracy; it's certainly possible for this to give a wrong answer if everything is scaled down by (say) 1e-9.

Reply:
Yeah, that's probably good.
I have pushed a commit which does this.